wasm2c/README.md

```diff
@@ -455,7 +458,7 @@ module doesn't use any globals, memory or tables.
 The most interesting part is the definition of the function `fac`:
 
 ```c
-static u32 w2c_fac(Z_fac_instance_t* instance, u32 w2c_p0) {
+static u32 w2c_fac_fac_0(w2c_fac* instance, u32 w2c_p0) {
```

Comment:
What is the `_0` suffix on the function name here?

Reply:
Oh I see we have an internal name and then an exported name that simply forwards its arg to the internal one.

As a followup should we potentially just export the function itself rather than a forwarding function?

Reply:
Yeah, exactly. In this case the names had to be deconflicted because the ID of the function itself is $fac, and also the export is named "fac", so the internal function gets the `_0` suffix.

I think trying to "directly" export a function like this may be doable but a bit complicated. :-( The internal functions are named and called based on their generated name or debug name (e.g. $f0 or $fac), whereas the exports are named based on their export name (in this case, also "fac"). The same function can be exported multiple times under different names, so the module could export this function as "fac" and "fac2" and "fac3" and they would all have to be in the .h file.

If you think this is valuable, I think what we'd try to do is detect a "direct" export (an exported function where the export name is identical to the name of the function itself) and handle it specially by declaring it directly in the `.h` file, and then for any other export of the same function, we just forward back to the original implementation. I can see a pathway here but it might be best done in another PR if you're amenable. :-/

Reply:
I yes, that makes total sense.  I think continue to separate internal names from export names makes sense give that context.   I don't think we need to special case for when they happen to match.

I do wonder if we could avoid declaring extra function though.. maybe function exports could be some kind of symbol alias instead?   I guess that runs into visibility issues, and is kind of compiler-specific.   Its a shame C doesn't really have anything like aliases built-in.

Reply:
Yeah, I think if we were willing to limit ourselves to GCC and clang, we could use `__attribute__((alias))` no problem. But I don't know how to do this on MSVC. :-/